typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
Yeah, I was bending over backwards trying not to re-implement `Room`'s `return_appearance` method because I really would've liked to keep `StoreRoom` tracking with any changes later developers make to `Room`. But I suppose this approach is just as vulnerable to changes if someone decides they want to represent room contents differently from how they're currently represented.

What about calling three component methods from Room's `return_appearance`:

1, Room description (can handle room properties like weather, darkness, etc. and can be shared with `StoreRoom`)
2. Exits (Can also be safely shared between `Room` and `StoreRoom`)
3. Contents (with StoreRoom overriding this method with its own custom implementation)

My thinking is the only thing that's unique about `StoreRoom`'s `return_appearance` is what it's doing with its contents display. So if we can inherit everything else from `Room`, that will keep it tracking with changes to `Room` (even new room appearance components) while providing the custom contents implementation?